Build failure: newlisp

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix run nixpkgs/nixos-unstable#newlisp
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318064161

### Relevant log output

```console
gcc -m64 -Wall -Wno-long-long -Wno-strict-aliasing -O2 -c  -DNEWCONFIG nl-import.c
nl-import.c: In function 'cdeclFunction':
nl-import.c:260:22: error: too many arguments to function 'function'; expected 0, have 1
  260 |             return  (*function)(args[0]);
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:263:22: error: too many arguments to function 'function'; expected 0, have 2
  263 |             return  (*function)(args[0], args[1]);
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:269:22: error: too many arguments to function 'function'; expected 0, have 3
  269 |             return  (*function)(args[0], args[1], args[2]);
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:272:22: error: too many arguments to function 'function'; expected 0, have 4
  272 |             return  (*function)(args[0], args[1], args[2], args[3]);
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:275:22: error: too many arguments to function 'function'; expected 0, have 5
  275 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:278:22: error: too many arguments to function 'function'; expected 0, have 6
  278 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:281:22: error: too many arguments to function 'function'; expected 0, have 7
  281 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:284:22: error: too many arguments to function 'function'; expected 0, have 8
  284 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:288:22: error: too many arguments to function 'function'; expected 0, have 9
  288 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:292:22: error: too many arguments to function 'function'; expected 0, have 10
  292 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:295:22: error: too many arguments to function 'function'; expected 0, have 11
  295 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:299:22: error: too many arguments to function 'function'; expected 0, have 12
  299 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:304:22: error: too many arguments to function 'function'; expected 0, have 13
  304 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
nl-import.c:309:22: error: too many arguments to function 'function'; expected 0, have 14
  309 |             return  (*function)(args[0], args[1], args[2], args[3],
      |                     ~^~~~~~~~~~ ~~~~~~~
make[1]: *** [makefile_build:17: nl-import.o] Error 1
make[1]: Leaving directory '/build/newlisp-10.7.5'
make: *** [Makefile:29: default] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.63, NixOS, 25.11 (Xantusia), 25.11.20251229.89dbf01`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/rv4z0vd7v0pkadinxd1c1xmd4b850ldr-source`

### Notify maintainers

I am the maintainer of the package `newlisp`.

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ne/newlisp/package.nix
+++ b/pkgs/by-name/ne/newlisp/package.nix
@@ -20,6 +20,9 @@ stdenv.mkDerivation (finalAttrs: {
     readline
   ];
 
+  # Build fails with C standard newer than C17
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
+
   configureScript = "./configure-alt";
 
   doCheck = true;

